src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/GetTaskByIdControllerTest.java

```diff
@@ -2,87 +2,227 @@
+    }
+
+    @Test
+    public void should_return_a_403_when_the_user_did_not_have_sufficient_permission_location_did_not_match() {
+
+        Map<String, String> task = common.setupTaskAndRetrieveIdsWithCustomVariable(LOCATION, "17595");
+
+        Response result = restApiActions.get(
+            "task/{task-id}",
+            task.get("taskId"),
+            authorizationHeadersProvider.getTribunalCaseworkerCAuthorization()
+        );
+
+        result.then().assertThat()
+            .statusCode(HttpStatus.FORBIDDEN.value())
+            .contentType(APPLICATION_JSON_VALUE)
+            .body("timestamp", lessThanOrEqualTo(LocalDateTime.now()
+                .format(DateTimeFormatter.ofPattern(DATE_TIME_FORMAT))))
+            .body("error", equalTo(HttpStatus.FORBIDDEN.getReasonPhrase()))
+            .body("status", equalTo(HttpStatus.FORBIDDEN.value()))
+            .body("message", equalTo(
+                format("User did not have sufficient permissions to access task with id: %s", task.get("taskId"))
+            ));
     }
 
+    @Test
+    public void should_return_a_200_and_retrieve_a_task_by_id_jurisdiction_region_and_location_match() {
 
-    private Map<String, String> setUpTaskWithCustomVariables(CamundaProcessVariables processVariables) {
         Map<String, String> task = common.setupTaskAndRetrieveIds();
-        given.iAddVariablesToTaskWithId(task.get("taskId"), processVariables);
-        return task;
+
+        Response result = restApiActions.get(
+            "task/{task-id}",
+            task.get("taskId"),
+            authorizationHeadersProvider.getTribunalCaseworkerDAuthorization()
+        );
+
+        result.then().assertThat()
+            .statusCode(HttpStatus.OK.value())
+            .and().contentType(MediaType.APPLICATION_JSON_VALUE)
+            .and().body("task.id", equalTo(task.get("taskId")));
     }
+
+    @Test
+    public void should_return_a_403_when_user_did_not_have_permission_jurisdiction_region_location_did_not_match() {
+        Map<String, String> task = common.setupTaskAndRetrieveIdsWithCustomVariable(LOCATION, "17595");
+
+        Response result = restApiActions.get(
+            "task/{task-id}",
+            task.get("taskId"),
+            authorizationHeadersProvider.getTribunalCaseworkerDAuthorization()
+        );
+
+        result.then().assertThat()
+            .statusCode(HttpStatus.FORBIDDEN.value())
+            .contentType(APPLICATION_JSON_VALUE)
+            .body("timestamp", lessThanOrEqualTo(LocalDateTime.now()
+                .format(DateTimeFormatter.ofPattern(DATE_TIME_FORMAT))))
+            .body("error", equalTo(HttpStatus.FORBIDDEN.getReasonPhrase()))
+            .body("status", equalTo(HttpStatus.FORBIDDEN.value()))
+            .body("message", equalTo(
+                format("User did not have sufficient permissions to access task with id: %s", task.get("taskId"))
+            ));
+    }
+
```

Comment:
Should we have tests that capture the 401 and 500 AC's? I.e.

- Then the endpoint will return a 401 HTTP error (not a valid user)

- Then the endpoint will return a 500 HTTP error (role assignment service error)